src/App.js

```diff
@@ -6,27 +6,53 @@ import Card from './components/UI/Card';
 class App extends React.Component {
   constructor(props) {
     super(props);
+
     this.state = {
+      checkboxUnderHeader: false,
       data: {
         inputValue: '',
         textareaValue: '',
       }
     }
   }
 
+  onCheckboxUnderHeaderChange = () => {
+    this.setState((state) => ({
+      checkboxUnderHeader: !state.checkboxUnderHeader,
+    }));    
+    console.log(this.state.checkboxUnderHeader)
+  }
+
   updateData = (tempData) => {
     this.setState({
       data: { ...tempData },
     });
+    
   }
 
   render() {
+    const {
+      data,
+      checkboxUnderHeader
+    } = this.state
+
+    const numbers = [1, 2, 3, 4, 5, 6, 7]
```

Comment:
Why random numbers? Declare the array of card data (with some values), pass with corresponding properties and use it to render array of cards.